src/java/org/apache/cassandra/io/sstable/indexsummary/IndexSummaryManager.java

```diff
@@ -296,6 +318,7 @@ public void shutdownAndWait(long timeout, TimeUnit unit) throws InterruptedExcep
             future.cancel(false);
             future = null;
         }
+        shutdownTasks.forEach(Runnable::run);
```

Comment:
what if `Runnable#run` throws? Then it will never get to ExecutorUtils.shutdownAndWait on the next line?